tinytuya/BulbDevice.py

```diff
@@ -136,36 +121,33 @@ def _rgb_to_hexvalue(r, g, b, bulb="A"):
 
         # Bulb Type A
         if bulb == "A":
-            # h:0-360,s:0-255,v:0-255|hsv|
             hexvalue = ""
-            for value in rgb:
-                temp = str(hex(int(value))).replace("0x", "")
-                if len(temp) == 1:
-                    temp = "0" + temp
-                hexvalue = hexvalue + temp
-
-            hsvarray = [int(hsv[0] * 360), int(hsv[1] * 255), int(hsv[2] * 255)]
-            hexvalue_hsv = ""
-            for value in hsvarray:
-                temp = str(hex(int(value))).replace("0x", "")
-                if len(temp) == 1:
-                    temp = "0" + temp
-                hexvalue_hsv = hexvalue_hsv + temp
-            if len(hexvalue_hsv) == 7:
-                hexvalue = hexvalue + "0" + hexvalue_hsv
-            else:
-                hexvalue = hexvalue + "00" + hexvalue_hsv
+
+            # r:0-255,g:0-255,b:0-255|rgb|
+            for rgb_value in rgb:
+                value = int(rgb_value)
+                if value < 0 or value > 255:
+                    raise ValueError(f"Bulb type {bulb} must have RGB values 0-255.")

```

Comment:
This is needed, but it occurs to me that we have a real mix of using python exceptions versus returning error payloads, or actually not doing any validation. This is more of a commentary on some of our global style/design we could address rather than specific to this PR. :)